token/cli/src/main.rs

```diff
@@ -1204,90 +1205,57 @@ async fn command_thaw(
     })
 }
 
-fn native_mint(program_id: &Pubkey) -> Result<Pubkey, Error> {
-    if program_id == &spl_token_2022::id() {
-        Ok(spl_token_2022::native_mint::id())
-    } else if program_id == &spl_token::id() {
-        Ok(spl_token::native_mint::id())
-    } else {
-        Err(format!("Error: unknown token program id {}", program_id).into())
-    }
-}
-
 async fn command_wrap(
     config: &Config<'_>,
     sol: f64,
     wallet_address: Pubkey,
     wrapped_sol_account: Option<Pubkey>,
+    immutable_owner: bool,
     bulk_signers: BulkSigners,
 ) -> CommandResult {
     let lamports = sol_to_lamports(sol);
+    let token = native_token_client_from_config(config);
 
-    let native_mint = native_mint(&config.program_id)?;
-    let instructions = if let Some(wrapped_sol_account) = wrapped_sol_account {
-        println_display(
-            config,
-            format!("Wrapping {} SOL into {}", sol, wrapped_sol_account),
-        );
-        vec![
-            system_instruction::create_account(
-                &wallet_address,
-                &wrapped_sol_account,
-                lamports,
-                Account::LEN as u64,
-                &config.program_id,
-            ),
-            initialize_account(
-                &config.program_id,
-                &wrapped_sol_account,
-                &native_mint,
-                &wallet_address,
-            )?,
-        ]
-    } else {
-        let account = get_associated_token_address_with_program_id(
+    let account = wrapped_sol_account.unwrap_or_else(|| {
+        get_associated_token_address_with_program_id(
             &wallet_address,
-            &native_mint,
+            token.get_address(),
             &config.program_id,
-        );
+        )
+    });
 
-        if !config.sign_only {
-            if let Some(account_data) = config
-                .rpc_client
-                .get_account_with_commitment(&account, config.rpc_client.commitment())
-                .await?
-                .value
-            {
-                if account_data.owner != system_program::id() {
-                    return Err(format!("Error: Account already exists: {}", account).into());
-                }
+    println_display(config, format!("Wrapping {} SOL into {}", sol, account));
+
+    if !config.sign_only {
+        if let Ok(account_data) = config.get_account_checked(&account).await {
+            if account_data.owner != system_program::id() {
+                return Err(format!("Error: Account already exists: {}", account).into());
```

Comment:
I'm a bit confused... `get_account_checked` makes sure that the owner of the account is `config.program_id`, so would we ever trigger the error? Seems like you might prefer to do a raw `get_account` rather than `get_account_checked`